feature-pitches/002-file-based-workflows/002-file-based-workflows.md

```diff
@@ -67,24 +63,16 @@ The file-based workflow feature should solve this. The workflow should read the
 
-The consequence of this is that there is no connection between iterations of a single workflow file as the user develops it. We recognize that this not *ideal* (for example, the user will only be able to compare `activities` generated by a non-modified `plan`; once they change the workflow, activities are not easily comparable anymore), however we consider it a worthwhile tradeoff for the sake of making workflow development an easy user experience.
-
-
-**How do we meaningfully list *relevant* workflow metadata with so many pieces of workflow metadata floating around?**
-
-The one cost of creating lots of workflow metadata is that listing the project's workflows yields a lot of metadata that the user doesn't care about. But the solution to that is right in the statement- the user doesn't care about any metadata that relate to data objects that no longer exist. `renku workflow ls` should only list workflows that relate to currently existing code and data objects, i.e. show me the workflow that I ran to create the current version of file `b`, not the workflow I ran yesterday that created a previous version of file `b`. (In the `plan` and `activity` vocabulary, only `plans` that have an `activity` that relates to a file currently in the project)
+* The workflows defined in files should be fully represented in the Renku KG metadata.
+* However, the workflow file itself should be the single source of truth for the workflow definition (not the KG metadata).
+* File based workflows can only be edited via the file. These workflows cannot be edited via the CLI or (in the future) somewhere in the RenkuLab UI.
+* In terms of versioning a file-based workflow, the metadata should contain metadata about the file-based workflow at each execution. There is no need to maintain a history of the file-based workflow at commits where the workflow was modified but not executed.
+* However, the KG must be aware of the latest state of the workflow file (even un-executed or not committed), because other parts of RenkuLab will be pulling information about the workflow via the KG (for example, the workflow browser will query the KG for the file-based workflow definition, and this needs to be up to date with the current state of the file so as not to confuse the user, even if the workflow file has not been executed since it was last modified).
+* It has been considered that there might be a need for a command to force-save the file-based workflow to the Renku metadata without executing it. For example, this might be helpful if the user modifies the description of the workflow, and wants to save this information to the KG without running their whole long-running workflow. However, the idea that there is a "save" beyond "file save" is a ugly area we'd like to avoid forcing upon the user if possible, so let's assume this is not needed. But the implementtion team should keep an eye out for other corner cases that might require such a command. 
```

Comment:
I think the implementation of such a command would be relatively easy, so it's a trade-off between the added complexity of having more commands and the potential need for such a command by users.